TSPL.docc/LanguageGuide/ErrorHandling.md

```diff
@@ -699,6 +699,204 @@ let photo = try! loadImage(atPath: "./Resources/John Appleseed.jpg")
+        counts[rating]! += 1
+    }
+
+    print("*", counts[1]!, "-- **", counts[2]!, "-- ***", counts[3]!)
+}
+```
+
+In the code above,
+the `summarize(_:)` function summarizes a list of ratings
+expressed on a scale of 1 to 3.
+This function throws an instance of `StatisticsError` if the input isn't valid.
+Both places in the code above that throw an error
+omit the type of the error
+because the function's error type is already defined.
+You can use the short form, `throw .noRatings`,
+instead of writing `throw StatisticsError.noRatings`
+when throwing an error in a function like this.
+
+When you write a specific error type at the start of the function,
+Swift checks that you don't throw any other errors.
+For example,
+if you tried to use `VendingMachineError` from examples earlier in this chapter
+in the `summarize(_:)` function above,
+that code would produce an error at compile time.
+
+You can call a function that uses typed throws
+from within a regular throwing function:
+
+```swift
+func someThrowingFunction() -> throws {
+    let ratings = [1, 2, 3, 2, 2, 1]
+    try summarize(ratings)
+}
+```
+
+The code above doesn't specify an error type for `someThrowingFunction()`,
+so it throws `any Error`.
+You could also write the error type explicitly as `throws(any Error)`;
+the code below is equivalent to the code above:
+
+```swift
+func someThrowingFunction() -> throws(any Error) {
+    let ratings = [1, 2, 3, 2, 2, 1]
+    try summarize(ratings)
+}
+```
+
+In this code,
+`someThrowingFunction()` propagates any errors that `summarize(_:)` throws.
+The errors from `summarize(_:)` are always `StatisticsError` values,
+which is also a valid error for `someThrowingFunction()` to throw.
+<!-- XXX Expand on subtyping here? -->
+
+In addition to specifying a function's error type,
+you can also write a specific error type for a `do`-`catch` statement.
+For example:
+
+```swift
+let ratings = []
+do throws(StatisticsError) {
+    try summarize(ratings)
+} catch {
+    switch error {
+    case .noRatings:
+        print("No ratings available")
+    case .invalidRating(let rating):
+        print("Invalid rating: \(rating)")
+    }
+}
+// Prints "No ratings available"
+```
+
+In this code,
+writing `do throws(StatisticsError)` indicates that
+the `do`-`catch` statement throws `StatisticsError` values as its errors.
+Like other `do`-`catch` statements,
+the `catch` clause can either handle every possible error
+or propagate unhandled errors for some surrounding scope to handle.
+This code handles all of the errors,
+using a `switch` statement with one case for each enumeration value.
+Like other `catch` clauses that don't have a pattern,
+the clause matches any error
+and binds the error to a local constant named `error`.
+Because the `do`-`catch` statement throws `StatisticsError` values,
+`error` is a value of type `StatisticsError`.
+
+<!-- XXX show multiple catch clauses with different patterns? -->
```

Comment:
I don't think you have to show multiple catch clauses specifically. It falls out from other sections.